cuda_core/cuda/core/experimental/_event.py

```diff
@@ -120,9 +120,27 @@ def __sub__(self, other):
         try:
             timing = handle_return(driver.cuEventElapsedTime(other.handle, self.handle))
         except CUDAError as e:
-            raise RuntimeError(
-                "Timing capability must be enabled in order to subtract two Events; timing is disabled by default."
-            ) from e
+            error_message = str(e)
+            if "CUDA_ERROR_INVALID_HANDLE" in error_message:
```

Comment:
I think instead of using `handle_return` above and working on the error message here we're be better off using the actual return code from the `driver.cuEventElapsedTime(other.handle, self.handle)` and constructing the message to raise with based on that instead?

Reply:
Maybe:

In _utils/cuda_utils.py, rename `_check_driver_error` (private name) → `raise_if_driver_error` (new and public name), then

```python
err, timing = driver.cuEventElapsedTime(other.handle, self.handle)
try:
    raise_if_driver_error(err) 
    return timing
except CUDAError as e:
    if err == driver.CUresult.CUDA_ERROR_INVALID_HANDLE:
        ....
    elif err == driver.CUresult. CUDA_ERROR_NOT_READY:
        .... 
    else:
        raise e
    raise RuntimeError(explanation) from e
```

Reply:
a72edac5930d668480df3c5239ddc34b89f6475f

Reply:
Do we want to re-raise `RuntimeError` or just the same `CUDAError`?

Reply:
nvm, we already raise `RuntimeError` before this PR